packages/ketcher-react/src/script/ui/dialog/template/TemplateDialog.tsx

```diff
@@ -325,23 +334,31 @@ const TemplateDialog: FC<Props> = (props) => {
 const selectTemplate = (template, props, dispatch) => {
   dispatch(selectTmpl(null))
   if (!template) return
+  dispatch(changeFilter(''))
   dispatch(selectTmpl(template))
   dispatch(onAction({ tool: 'template', opts: template }))
   props.onOk(template)
 }
 
+const exit = (props, dispatch) => {
```

Comment:
Propose to rename this to something like `onModalCancel`, `onClose`, `onModalClose`...